### DISCUSSION TITLE
New package for LuaLaTeX and XeLaTeX users

### DISCUSSION THREAD

**Discussion by psb1558** [psb1558/Junicode-font#258] [General]
I have just uploaded a tiny package (script?) for loading Junicode (static version) in LuaLaTeX and XeLaTeX documents. It is meant to simplify loading the font, provide options for customizing the style, and provide a number of commands to load different styles and simplify the use of Junicode's many Stylistic Sets and Character Variant features.

The package is accompanied by files that serve as both documentation and example. My plans are to

1. write a separate script for the variable version of Junicode, which is of course much more in need of simplifying than the static version, and
2. include this with Junicode 2.204 or some other future release.

I'd be very interested in impressions and suggestions from persons who use Junicode with LuaLaTeX and XeLaTeX. Find the package in the `TeX` directory of the repository.

**Comment by Doc73** [comment]
If I'm not being indiscreet, are you planning to upload the new version to CTAN?

**Reply by psb1558** [reply]
CTAN is out of the Junicode business for the time being: they took their package down because it was so out of date (my fault), and releases are too frequent right now to keep up with. (I was part of the decision-making about this.) If the situation stabilizes I'll submit a new package, or get someone to do it. But for right now I'm suggesting that people just drop Junicode.sty into their working directory and let LuaTex/XeTeX find the font installed on their systems.

**Reply by alerque** [reply]
I automated CTAN releases for Libertinus. FontShip now includes tooling to do this with very minimal user intervention to get CTAN updated. If you want to open an issue for this and assign it to me I can look into setting up FontShip for this project. I won't get to it right away, but with an assigned issue I might someday ;-)

**Reply by psb1558** [reply]
Thanks, @alerque: I will do that. Want to emphasize that there is no hurry at all about this.

**Comment by psb1558** [comment]
I have added a package, JunicodeVF.sty, for the variable font. This works only in LuaLaTeX, since other flavors of TeX do not handle variable fonts. It is a bit more involved to use than the package for the static font (though it is useful *out of the box*, as they say). I hope to make it a bit more convenient, but first I'm going to have to learn more about LaTeX3 programming, which I find quite challenging. There is a doc to go with it.

**Comment by jsbien** [comment]
I don't use Junicode for the whole documents, just for examples. At first glance it seems the package is not for me. Am I right?

**Comment by jsbien** [comment]
I tried to typeset junicode-package.tex and got several errors, cf. the enclosed log (the missing font problem is probably my fault).
[junicode-package.zip](https://github.com/psb1558/Junicode-font/files/13797582/junicode-package.zip).

**Comment by psb1558** [comment]
This is odd. The errors start with XeLaTeX not recognizing conditionals created by `\newif`, and then all the other errors are triggered by that. Meanwhile the script works fine in LuaLaTeX. (I tested it a while ago in XeLaTeX, but something I've done since then has broken it.)

This may take some study to figure out.

**Comment by psb1558** [comment]
I've uploaded a fixed (I hope) version of junicode.sty. There seem to be varioius subtle differences in the way XeTeX and LuaTeX process things, and I ran afoul of those. The one remaining problem with this document (on my system anyway) is that it tries to load the fonts Fira Mono and Fira Sans (both in the TeXLive distribution), but does not find them. If you run into this problem, you can either install the Fira fonts in your system or substitute other fonts for them.

**Reply by jsbien** [reply]
It works now, thanks. No Fira font problems, but I had to install SourceCodePro-Regular and SourceSansPro-Regular. Just curious, what are the advantages of these fonts?

**Comment by psb1558** [comment]
I thought I had changed SourceCodePro to Fira, but I guess I forgot that. One doesn't have to substitute for the default mono and sans fonts, but I do like SourceCodePro, though it's a bit too wide to harmonize with Junicode (Fira works better). I also like loading fonts with fontspec so I can scale them with `Scale=MatchLowercase`, which makes mono textl look a good bit better.

**Reply by jsbien** [reply]
Thanks for the explanation. You have changed to Fira only in `junicodevf-package.tex`.

In `junicode-package.tex` the numbers are old style by default. `tabular` doesn't change it, and there is no information in the documentation how to switch this off. I used `\addfontfeature{Numbers={Uppercase,Monospaced}}` but I guess there is a best way.
